### PR TITLE
⬆️🔧🩹🐛 media-libs/libavif: add 1.1.1, fix 9999

### PR DESCRIPTION
**Pull request by Jamim**
_Hello everyone,_

**libavif** `1.1.1` [was released](https://github.com/AOMediaCodec/libavif/releases/tag/v1.1.1) recently.
It would be nice to have it in the Portage tree.

Based on:
  - [media-libs/libavif-1.0.4](https://gitweb.gentoo.org/repo/gentoo.git/tree/media-libs/libavif/libavif-1.0.4.ebuild?id=dc2da66d) from #35356 by @novomesk
  - ~AOMediaCodec/libavif#2263~
  - ~AOMediaCodec/libavif#2301~

Changes in comparison to `1.0.4`:
  - 🔧 update configuration options
  - 🩹 fix missing `media-gfx/imagemagick[lcms]`
    for the `test_cmd_icc_profile` test

<details><summary>diff -u libavif-1.{0.4,1.1}.ebuild</summary>
<p>

```diff
--- libavif-1.0.4.ebuild	2024-04-16 23:26:12.922080278 +0300
+++ libavif-1.1.1.ebuild	2024-07-31 10:54:33.040342797 +0300
@@ -25,7 +25,12 @@
 	media-libs/libpng:=[${MULTILIB_USEDEP}]
 	aom? ( >=media-libs/libaom-3.3.0:=[${MULTILIB_USEDEP}] )
 	dav1d? ( >=media-libs/dav1d-1.0.0:=[${MULTILIB_USEDEP}] )
-	extras? ( test? ( dev-cpp/gtest ) )
+	extras? (
+		test? (
+			dev-cpp/gtest
+			media-gfx/imagemagick[lcms]
+		)
+	)
 	gdk-pixbuf? (
 		dev-libs/glib:2[${MULTILIB_USEDEP}]
 		x11-libs/gdk-pixbuf:2[${MULTILIB_USEDEP}]
@@ -43,13 +48,16 @@
 multilib_src_configure() {
 	local mycmakeargs=(
 		-DBUILD_SHARED_LIBS=ON
-		-DAVIF_CODEC_AOM=$(usex aom ON OFF)
-		-DAVIF_CODEC_DAV1D=$(usex dav1d ON OFF)
 		-DAVIF_CODEC_LIBGAV1=OFF
 
+		# bug 916948
+		-DAVIF_LIBYUV=OFF
+
 		# Use system libraries.
-		-DAVIF_LOCAL_ZLIBPNG=OFF
-		-DAVIF_LOCAL_JPEG=OFF
+		-DAVIF_CODEC_AOM=$(usex aom SYSTEM OFF)
+		-DAVIF_CODEC_DAV1D=$(usex dav1d SYSTEM OFF)
+		-DAVIF_ZLIBPNG=SYSTEM
+		-DAVIF_JPEG=SYSTEM
 
 		-DAVIF_BUILD_GDK_PIXBUF=$(usex gdk-pixbuf ON OFF)
 
@@ -58,13 +66,14 @@
 
 	if multilib_is_native_abi; then
 		mycmakeargs+=(
-			-DAVIF_CODEC_RAV1E=$(usex rav1e ON OFF)
-			-DAVIF_CODEC_SVT=$(usex svt-av1 ON OFF)
+			-DAVIF_CODEC_RAV1E=$(usex rav1e SYSTEM OFF)
+			-DAVIF_CODEC_SVT=$(usex svt-av1 SYSTEM OFF)
 
 			-DAVIF_BUILD_EXAMPLES=$(usex examples ON OFF)
 			-DAVIF_BUILD_APPS=$(usex extras ON OFF)
 			-DAVIF_BUILD_TESTS=$(usex test ON OFF)
 			-DAVIF_ENABLE_GTEST=$(usex extras $(usex test ON OFF) OFF)
+			-DAVIF_GTEST=$(usex extras $(usex test SYSTEM OFF) OFF)
 		)
 	else
 		mycmakeargs+=(
@@ -75,6 +84,7 @@
 			-DAVIF_BUILD_APPS=OFF
 			-DAVIF_BUILD_TESTS=OFF
 			-DAVIF_ENABLE_GTEST=OFF
+			-DAVIF_GTEST=OFF
 		)
 
 		if ! use aom ; then
```

</p>
</details> 

```
100% tests passed, 0 tests failed out of 48

Total Test time (real) =  12.76 sec
 * Tests succeeded.
```

Fixed in `9999`:
  - 🔧 configuration options
  - 🩹 missing `media-gfx/imagemagick[lcms]`
    for the `test_cmd_icc_profile` test
  - ~🐛 missing CXX for tests~

Updates to `metadata.xml:`
  - add `bugs-to` url

_Best regards!_

---

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

### CODE REVIEW

File: media-libs/libavif/libavif-9999.ebuild
Comment (novomesk) [inline media-libs/libavif/libavif-9999.ebuild:45]: Is this patch still needed in live ebuild after
https://github.com/AOMediaCodec/libavif/commit/3640a3ae3dd0082301c43b3d4d182714a06662d8
?

Comment (Jamim) [reply media-libs/libavif/libavif-9999.ebuild]: Hello @novomesk,
No, it's not needed anymore.
I've dropped it in 3af4480.

**Comment by Jamim** [comment]
_Hello @novomesk,_
Can this PR be approved now?
_Thanks!_

**Comment by novomesk** [comment]
Look fine but I can't merge.

**Comment by Jamim** [comment]
> _@novomesk_
> Look fine but I can't merge.

_Hello @mgorny,_
Would you mind helping us merge these changes?
_Thanks in advance!_

**Review by juippis — APPROVED** [review]
```
 FILES:-usr/lib64/libavif.so.16.0.4
 FILES:+usr/lib64/libavif.so.16.1.1
   ABI: libavif.so.16(64) func(+1)
```
https://bugs.gentoo.org/822336#c1

I'll bump the subslot. 

Also could you stop pinging people on PRs that are **assigned** to proxy-maint project? The team will get there in time. For PRs that are **not** assigned to an active project, you may request help reviewing usually in IRC (#gentoo-dev-help, #gentoo-proxy-maint).

**Comment by Jamim** [comment]
_Hello @juippis,_

Thank you for taking care of this PR! I really appreciate this!

> Also could you stop pinging people on PRs that are assigned to proxy-maint project?

TL;DR: I'm not sure.

I had a feeling like I had to answer something, but for a week or so I was failing to formulate my thoughts and say something that won't insult anyone.

I suppose you know how frustrating contributing to open source might be from time to time.
And contributing to Gentoo is frustrating most of the time, so I'm not even sure why [I do it](https://github.com/gentoo/gentoo/pulls?q=sort%3Aupdated-desc+is%3Apr+author%3AJamim).

Of course, I understand that everyone is busy with their work, life, and no matter what else more important than reviewing and merging those damn PRs. That's fair. So I try hard to only ping people in case I lose hope for a certain PR. And I try to do my best to find a person who is somehow related. For instance, here I asked **Michał Górny** because he was the only person who committed changes to [media-libs/libavif](https://gitweb.gentoo.org/repo/gentoo.git/log/media-libs/libavif) since February.

> The team will get there in time.

🙄

> Also thanks, good work here. Lots of CMake related changes and everything looks fine.

❤️

**Comment by juippis** [comment]
> 
> TL;DR: I'm not sure.
> 
> I had a feeling like I had to answer something, but for a week or so I was failing to formulate my thoughts and say something that won't insult anyone.

Appreciate that, it's a step towards healthy relationship :)

> 
> I suppose you know how frustrating contributing to open source might be from time to time. And contributing to Gentoo is frustrating most of the time, so I'm not even sure why [I do it](https://github.com/gentoo/gentoo/pulls?q=sort%3Aupdated-desc+is%3Apr+author%3AJamim).

I know. And I'll admit there are many inactive projects who don't even look at Github PRs - luckily your open PR count isn't that bad. The team can maybe help you with inactive projects, but proxy-maint takes priority and not all projects welcome "outside" contributors - so we can't merge stuff that belongs to e.g. Gnome.

> 
> Of course, I understand that everyone is busy with their work, life, and no matter what else more important than reviewing and merging those damn PRs. That's fair. So I try hard to only ping people in case I lose hope for a certain PR. 

What I'm trying to say is, the proxy-maint team **loves** to see and merge these PRs. And we all get notifications for these PRs already. You pinging random devs just adds more noise and more e-mails, ultimately I believe making people ignore you rather than answer your call. Can you imagine if **every** contributor started pinging everyone after few days of silence? It'd make following these PRs impossible, and in the end, us limiting the pinging ability to project members only. I will admit the proxy-maint has been lagging behind a lot, but things are really starting to look good again for maintainer-needed and proxy-maint packages. 

If you do find the process unjoyful, then at least take a break. After all it's voluntary work, you should never feel bad about it. I promise things are looking better now, but it may take 1-2 weeks to fully get caught up with project work and process few older PRs.

**Pull request closed without merging**
